src/tf/tf.cpp

```diff
@@ -363,6 +363,15 @@ struct tf_parser
         int64_t axis = 0;
         if(contains(attributes, "axis"))
             axis = attributes.at("axis").i();
+        size_t input_size = args.front()->get_shape().lens().size();
+        if(axis > input_size)
+        {
+            MIGRAPHX_THROW("Error in protobuf: axis must be smaller than input size");
+        }
+        // check if input arg needs axis to be converted to NCHW
+        if(input_size >= 4)
+            axis = parse_axis(axis);
```

Comment:
Does `parse_axis` work for 5 dimensions?

Reply:
Yes, it should not modify any of the dim values above the 5th dimension.

Reply:
We may want to change the `parse_axis` function to take the dims, because when we do 3d convolutions then we will need to handle 5 dims.

Reply:
Do we want to open a separate issue for this? There are several operators which use parse_axis.

Reply:
Yea probably.